lib/game.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class Game
+  def initialize(_player1, _player2)
+    # @current_player_1 = player1
+    # @current_player_2 = player2
+    @player1_played = []
+    @player2_played = []
+    @board_array = []
+    @box_played = []
+    @game_played = []
+    @ans = false
+    board
+  end
+
+  def board
+    (1..9).each do |i|
+      @board_array.push(" [ #{i} ] ").each_slice(3)
+    end
+    @board_array.each_slice(3) { |x| puts x.join puts }
+  end
+
+  def cheater?(box)
+    @ans = false
+    if @game_played.include?(box)
+      @ans = true
+    else
+      @game_played.push(box)
+    end
+    @ans
+  end
+
+  def play(player, box, current_player)
+    if current_player == 1
+      player_played = @player1_played
+      sym = 'X'
+    elsif current_player == 2
+      player_played = @player2_played
+      sym = 'O'
+    end
+    check(player, player_played, box, sym)
+  end
+
+  def check(player, player_played, box, sym)
+    player_played.push(box)
+    @winning_combos = [[1, 2, 3], [4, 5, 6], [7, 8, 9], [1, 4, 7], [2, 5, 8], [3, 6, 9], [1, 5, 9], [3, 5, 7]]
+    (0...@winning_combos.length).each do |i|
+      if (@winning_combos[i] - player_played).empty?
+        @announce = "YOU WON #{player.upcase}!!"
+        @result = true
+        break
+      elsif @box_played.length == 8
+        @announce = "IT'S A TIE"
+        @result = true
+        break
+      else
+        @result = false
+      end
+    end
+    @box_played.push(box)
+    @board_array[box - 1] = " [ #{sym} ] "
+    puts @board_array.each_slice(3) { |x| puts x.join puts }
```

Comment:
Let's avoid the use of puts/prints in classes other than `bin/main.rb`

This ensures keeping user interaction out of the game logic.

Reply:
@bolah2009, this was only to create the board for the game and felt it would be less hectic to keep it controlled within the game logic instead of outputting it on the main.... I don't know if I explained that correctly.. Is there a method you could recommend us?

Reply:
What I'm trying to say is, that's not technically a user interaction, but updating the user input within the game logic and outputting it.

Reply:
The user has no direct interaction with that part of the game logic, but it's an update of their input and displaying the updated board.

Reply:
I just said the same thing three different ways. Let me know if you still find it confusing.

Reply:
Here's a fourth way -- That's not a _user interaction_; it's _part of the board display._

Reply:
But the important reason why we kept this as is, is because it goes with the flow of the game. It doesn't get called independently, but updates automatically.

Reply:
Hey @yirano , I understand your point. Using `puts/gets` in anyway is user interaction. Both input and output is user interaction. I don't know if you have seen the example above (I edited the comment).
We could use something like that to update the board and return a `String`. Let `main.rb` outputs it to the user.

Reply:
Darn it. Okay! We think we've come up with a way to display the word without using puts.

Reply:
[unrelated] You are the only one making commits, please remember this is a collaborative project. Let's make sure there's commits from both collaborators. :blush:

Reply:
Oh! That's because I commit for every little thing. It's a weird habit that I need to stop.....
Delete a line - commit
Get rid of a white space - commit
Comment out a test case even. Those don't make sense but I do those too for some reason. 😆